backend/src/core/services/stats.service.ts

```diff
@@ -158,6 +158,45 @@ const getCurrentStats = async (
     halted: job.campaign.halted,
     status_updated_at: job.updatedAt, // Timestamp when job was logged
   }
+
+  // If job is not in send queue, return archived stats
+  if (job.status !== JobStatus.Ready) {
+    return stats
+  }
+
+  const campaign = await Campaign.findByPk(campaignId)
+  if (!campaign || !campaign.credName) {
```

Comment:
The existence of the campaign is checked in isEmailCampaignOwnedByUser, isSmsCampaignOwnedByUser and isTelegramCampaignOwnedByUser

Reply:
And it must have had a non-null credName for a job to be created